staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1018,7 +1018,6 @@ func validatePatchWithSetOrderList(patchList, setOrderList interface{}, mergeKey
 	if patchIndex < len(nonDeleteList) && setOrderIndex >= len(typedSetOrderList) {
 		return fmt.Errorf("The order in patch list:\n%v\n doesn't match %s list:\n%v\n", typedPatchList, setElementOrderDirectivePrefix, setOrderList)
 	}
-	typedPatchList = append(nonDeleteList, toDeleteList...)
```

Comment:
typedPatchList is referenced in line 1019. Maybe it's a bug of the linter.

On the other hand, this line is problematic, it's recommended that the result of append is only assigned to the first argument itself: https://stackoverflow.com/questions/35276022/unexpected-slice-append-behaviour.

Reply:
Thanks a lot for your feedback @caesarxuchao! Wasn't aware of append's behaviour described in the stackoverflow question.
I think it's fine to remove `typedPatchList` here. The value which is being assigned isn't used below.

Reply:
Oh, you are right. I thought the code block was in a for-loop but it wasn't. The linter is correct after all.